qubes/storage/zfs_encrypted.py

```diff
@@ -78,12 +79,30 @@ def __repr__(self):
         try:
             # TODO how do we pass $1 to this stuff? we can pass **kwargs to
             # asyncio.create_subprocess_exec, but we can't influence command
-            await pw_vm.run_service_for_stdio(
-                'qubes.AskPassword',
-                autostart=True, gui=True,
-                user='user',
-                input=self.name.encode()+b'\n', # context for the prompt
-                stdout=pw_pipe_out)
+            # await pw_vm.run_service_for_stdio(
+            # TODO THIS used to work, now it fucking broke.
```

Comment:
If you direct dom0, it needs R4.1 commit https://github.com/QubesOS/qubes-core-admin/commit/50adc6088236aaa18ad12ce54401a770a0d554de. If you want to test on R4.0, you need to backport this commit.
Or maybe some other problem?

Reply:
That seems likely. It used to work on 4.0, but after updating earlier this week it stopped working with the exception that `run_service_for_stdio` for the `AdminVM` class was not defined.